### PR TITLE
Main into v3

### PR DESCRIPTION
**Pull request by tdchow**
### Summary of changes

 - Merge main into v3

 ### Checklist

 - [ ] Added a changelog entry

-

### CODE REVIEW

File: demo/src/main/java/com/braintreepayments/api/browserswitch/demo/DemoActivity.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+//import androidx.appcompat.app.AppCompatActivity;
+//import androidx.fragment.app.Fragment;
+//import androidx.fragment.app.FragmentManager;
+//
+//import com.braintreepayments.api.BrowserSwitchClient;
+//import com.braintreepayments.api.BrowserSwitchException;
+//import com.braintreepayments.api.BrowserSwitchOptions;
+//import com.braintreepayments.api.BrowserSwitchResult;
+//
+//public class DemoActivity extends AppCompatActivity {
```
Comment (saperi22) [inline demo/src/main/java/com/braintreepayments/api/browserswitch/demo/DemoActivity.java:18]: should this be removed?

Comment (tdchow) [reply demo/src/main/java/com/braintreepayments/api/browserswitch/demo/DemoActivity.java]: yep 😅 just removed it.

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,17 +4,21 @@
...
+    * Bump target Java version to Java 11
+    * Upgrade Kotlin version to 1.9.10
+    * Upgrade to Android Gradle Plugin 8
+    * Change `BrowserSwitchClient#start` parameters and return type
+    * Change `BrowserSwitchClient#parseResult` parameters
+    * Remove `deliverResult`, `getResult`, `captureResult`, `clearActiveRequests`, `getResultFromCache`, and `deliverResultFromCache` from `BrowserSwitchClient`
+    * Add `BrowserSwitchRequest` and `BrowserSwitchPendingRequest`
+    * Convert `BrowserSwitchResult` to sealed class and add `BrowserSwitchResultInfo`
+    * Remove `BrowserSwitchStatus`
+    * Rename `parseResult()` to `completeRequest()`
```
Comment (jaxdesmarais) [inline CHANGELOG.md:17]: Looks like these are bumped in a little too far now

File: browser-switch/src/test/java/com/braintreepayments/api/BrowserSwitchClientUnitTest.java [hunk 3]
```diff
@@ -170,21 +178,51 @@ public void start_whenNoReturnUrlSchemeSet_throwsFailure() {
...
                 .url(browserSwitchDestinationUrl)
                 .metadata(metadata);
         BrowserSwitchPendingRequest request = sut.start(componentActivity, options);
         assertTrue(request instanceof BrowserSwitchPendingRequest.Failure);
-        assertEquals("A returnUrlScheme is required.", ((BrowserSwitchPendingRequest.Failure) request).getCause().getMessage());
+        assertEquals("An appLinkUri or returnUrlScheme is required.", ((BrowserSwitchPendingRequest.Failure) request).getCause().getMessage());
     }
 
+
+    // TODO: FIX THIS
```
Comment (jaxdesmarais) [inline browser-switch/src/test/java/com/braintreepayments/api/BrowserSwitchClientUnitTest.java:190]: I assume this should still be commented out?

**Review by sarahkoop — APPROVED** [review]
One commented out test Jax already noted, otherwise LGTM

File: CHANGELOG.md (hunk 2 above)
Comment (tdchow) [reply CHANGELOG.md]: fixed!

File: browser-switch/src/test/java/com/braintreepayments/api/BrowserSwitchClientUnitTest.java (hunk 3 above)
Comment (tdchow) [reply browser-switch/src/test/java/com/braintreepayments/api/BrowserSwitchClientUnitTest.java]: i ended up fixing the test case 👍

**Review by sshropshire — APPROVED** [review]

**Review by jaxdesmarais — APPROVED** [review]

**Pull request merged**